Issues with building the knowledge base

Pattern-3


###################################
# wso2am-publisher / wso2am-store #
###################################


repository/components/api-publisher/configuration/org.eclipse.equinox.simpleconfigurator/bundles.info
=====================================================================================================

Why following two lines are added?
@@ -239,6 +239,8 @@
 org.wso2.carbon.core.services,4.4.7,../plugins/org.wso2.carbon.core.services_4.4.7.jar,4,true
 org.wso2.carbon.databridge.agent,5.1.0,../plugins/org.wso2.carbon.databridge.agent_5.1.0.jar,4,true
 org.wso2.carbon.databridge.commons,5.1.0,../plugins/org.wso2.carbon.databridge.commons_5.1.0.jar,4,true
+org.wso2.carbon.databridge.commons.binary,5.1.0,../plugins/org.wso2.carbon.databridge.commons.binary_5.1.0.jar,4,true
+org.wso2.carbon.databridge.commons.thrift,5.1.0,../plugins/org.wso2.carbon.databridge.commons.thrift_5.1.0.jar,4,true
 org.wso2.carbon.deployment.synchronizer,4.5.4,../plugins/org.wso2.carbon.deployment.synchronizer_4.5.4.jar,4,true
 org.wso2.carbon.deployment.synchronizer.subversion,4.5.4,../plugins/org.wso2.carbon.deployment.synchronizer.subversion_4.5.4.jar,4,true
 org.wso2.carbon.directory.server.manager,5.2.0,../plugins/org.wso2.carbon.directory.server.manager_5.2.0.jar,4,true


repository/conf/axis2/axis2.xml
===============================

All hostnames are allowed.
@@ -543,7 +543,7 @@
                 <Password>wso2carbon</Password>
             </TrustStore>
         </parameter>
-        <!--<parameter name="HostnameVerifier">DefaultAndLocalhost</parameter>-->
+        <parameter name="HostnameVerifier">AllowAll</parameter>
             <!--supports Strict|AllowAll|DefaultAndLocalhost or the default if none specified -->
     </transportSender>


repository/conf/api-manager.xml
===============================

No link between publisher and store
@@ -628,7 +628,7 @@
      getting this node to join the cluster.
      -->
     <clustering class="org.wso2.carbon.core.clustering.hazelcast.HazelcastClusteringAgent"
-                enable="false">
+                enable="true">

         <!--
            This parameter indicates whether the cluster has to be automatically initalized
@@ -650,7 +650,7 @@
                     is deemed to have left the cluster, it will be detected by the Group Membership
                     Service (GMS) using a TCP ping mechanism.
         -->
-        <parameter name="membershipScheme">multicast</parameter>
+        <parameter name="membershipScheme">wka</parameter>
         <!--<parameter name="licenseKey">xxx</parameter>-->
         <!--<parameter name="mgtCenterURL">http://localhost:8081/mancenter/</parameter>-->

@@ -658,7 +658,7 @@
          The clustering domain/group. Nodes in the same group will belong to the same multicast
          domain. There will not be interference between nodes in different groups.
         -->
-        <parameter name="domain">wso2.carbon.domain</parameter>
+        <parameter name="domain">wso2.pub.store.domain</parameter>

         <!-- The multicast address to be used -->
         <!--<parameter name="mcastAddress">228.0.0.4</parameter>-->
@@ -679,7 +679,7 @@
         -->
         <!-- The host name or IP address of this member -->

-        <parameter name="localMemberHost">127.0.0.1</parameter>
+        <parameter name="localMemberHost">publisher</parameter>

         <!--
             The bind adress of this member. The difference between localMemberHost & localMemberBindAddress
@@ -720,7 +720,7 @@
         -->
         <members>
             <member>
-                <hostName>127.0.0.1</hostName>
+                <hostName>store</hostName>
                 <port>4000</port>
             </member>
         </members>

Why there is no link between publisher and key manager?
@@ -16,7 +16,7 @@
          logic relies on this. -->
     <AuthManager>
         <!-- Server URL of the Authentication service -->
-        <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>
+        <ServerURL>https://keymanager:9443/services/</ServerURL>
         <!-- Admin username for the Authentication manager. -->
         <Username>${admin.username}</Username>
         <!-- Admin password for the Authentication manager. -->

Same with follow components
@@ -87,13 +87,13 @@
                 <Name>Production and Sandbox</Name>
                 <Description>This is a hybrid gateway that handles both production and sandbox token traffic.</Description>
                 <!-- Server URL of the API gateway -->
-                <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>
+                <ServerURL>https://gateway-manager:9443/services/</ServerURL>
                <!-- Admin username for the API gateway. -->
                 <Username>${admin.username}</Username>
                 <!-- Admin password for the API gateway.-->
                 <Password>${admin.password}</Password>
                 <!-- Endpoint URLs for the APIs hosted in this API gateway.-->
-                <GatewayEndpoint>http://${carbon.local.ip}:${http.nio.port},https://${carbon.local.ip}:${https.nio.port}</GatewayEndpoint>
+                <GatewayEndpoint>http://gateway-worker:8280,https://gateway-worker:8243</GatewayEndpoint>
             </Environment>
         </Environments>
     </APIGateway>

@@ -202,7 +202,7 @@
     -->
     <APIKeyValidator>
         <!-- Server URL of the API key manager -->
-        <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>
+        <ServerURL>https://keymanager:9443/services/</ServerURL>
 
         <!-- Admin username for API key manager. -->
         <Username>${admin.username}</Username>

@@ -216,7 +216,7 @@
              -If you need to start two API Manager instances in the same machine, you need to give different ports to "ThriftServerPort" value in two nodes.
              -ThriftServerHost - Allows to configure a hostname for the thrift server. It uses the carbon hostname by default.
             -The Gateway uses this parameter to connect to the key validation thrift service. -->
-        <KeyValidatorClientType>ThriftClient</KeyValidatorClientType>
+        <KeyValidatorClientType>WSClient</KeyValidatorClientType>
         <ThriftClientConnectionTimeOut>10000</ThriftClientConnectionTimeOut>
         <!--ThriftClientPort>10397</ThriftClientPort-->

@@ -284,7 +284,7 @@
         <!--This property is used to indicate how we do user name comparision for token generation https://wso2.org/jira/browse/APIMANAGER-2225-->
         <CompareCaseInsensitively>true</CompareCaseInsensitively>
         <DisplayURL>false</DisplayURL>
-        <URL>https://localhost:${mgt.transport.https.port}/store</URL>
+        <URL>https://store:9446/store</URL>
 
         <!-- Server URL of the API Store. -->
         <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>

@@ -479,7 +479,7 @@
             </DataPublisherThreadPool>
         </DataPublisher>
         <PolicyDeployer>
-            <ServiceURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServiceURL>
+            <ServiceURL>https://traffic-manager:9443/services/</ServiceURL>
             <Username>${admin.username}</Username>
             <Password>${admin.password}</Password>
         </PolicyDeployer>

Why follow change?
@@ -484,12 +484,12 @@        
             <Password>${admin.password}</Password>
         </PolicyDeployer>
         <BlockCondition>
-            <Enabled>true</Enabled>
+            <Enabled>false</Enabled>
             <!--InitDelay>300000</InitDelay>
             <Period>3600000</Period-->
         </BlockCondition>
         <JMSConnectionDetails>
-            <Enabled>true</Enabled>
+            <Enabled>false</Enabled>
             <ServiceURL>tcp://${carbon.local.ip}:${jms.port}</ServiceURL>
             <Username>${admin.username}</Username>
             <Password>${admin.password}</Password>


repository/conf/carbon.xml
==========================

Why follow change?
@@ -582,11 +582,11 @@
     -->
     <DeploymentSynchronizer>
         <Enabled>false</Enabled>
-        <AutoCommit>false</AutoCommit>
+        <AutoCommit>true</AutoCommit>
         <AutoCheckout>true</AutoCheckout>
         <RepositoryType>svn</RepositoryType>
-        <SvnUrl>http://svnrepo.example.com/repos/</SvnUrl>
-        <SvnUser>username</SvnUser>
+        <SvnUrl>http://svnrepo/repo</SvnUrl>
+        <SvnUser>user</SvnUser>
         <SvnPassword>password</SvnPassword>
         <SvnUrlAppendTenantId>true</SvnUrlAppendTenantId>
     </DeploymentSynchronizer>


repository/conf/jndi.properties
===============================

No link between publisher and traffic manager
@@ -16,9 +16,9 @@

 # register some connection factories
 # connectionfactory.[jndiname] = [ConnectionURL]
-connectionfactory.TopicConnectionFactory = amqp://admin:admin@clientid/carbon?brokerlist='tcp://localhost:5672'
+connectionfactory.TopicConnectionFactory = amqp://admin:admin@clientid/carbon?brokerlist='tcp://traffic-manager:5672'

-connectionfactory.QueueConnectionFactory = amqp://admin:admin@clientID/test?brokerlist='tcp://localhost:5672'
+connectionfactory.QueueConnectionFactory = amqp://admin:admin@clientID/test?brokerlist='tcp://traffic-manager:5672'

 # register some queues in JNDI using the form
 # queue.[jndiName] = [physicalName]


repository/resources/security/client-truststore.jks, repository/resources/security/wso2cabon.jks
================================================================================================
    Content has been changed but cannot read.


######################
# wso2am-key_manager #
######################


repository/conf/api-manager.xml
================================

No direct link between keymanager and gateway-manager / gateway-worker
@@ -87,13 +87,13 @@
                 <Name>Production and Sandbox</Name>
                 <Description>This is a hybrid gateway that handles both production and sandbox token traffic.</Description>
                 <!-- Server URL of the API gateway -->
-                <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>
+                <ServerURL>https://gateway-manager:9443/services/</ServerURL>
                <!-- Admin username for the API gateway. -->
                 <Username>${admin.username}</Username>
                 <!-- Admin password for the API gateway.-->
                 <Password>${admin.password}</Password>
                 <!-- Endpoint URLs for the APIs hosted in this API gateway.-->
-                <GatewayEndpoint>http://${carbon.local.ip}:${http.nio.port},https://${carbon.local.ip}:${https.nio.port}</GatewayEndpoint>
+                <GatewayEndpoint>http://gateway-worker:8280,https://gateway-worker:8243</GatewayEndpoint>
             </Environment>
         </Environments>
     </APIGateway>

No link to analytics
@@ -136,20 +136,20 @@
 
              Ex - Multiple Receiver Groups with two receivers each
              {tcp://localhost:7612/,tcp://localhost:7613},{tcp://localhost:7712/,tcp://localhost:7713/} -->
-        <DASServerURL>{tcp://localhost:7612}</DASServerURL>
-        <!--DASAuthServerURL>{ssl://localhost:7712}</DASAuthServerURL-->
+        <DASServerURL>tcp://analytics:7612</DASServerURL>
+        <DASAuthServerURL>ssl://analytics:7712</DASAuthServerURL>
         <!-- Administrator username to login to the remote DAS server. -->
-        <DASUsername>${admin.username}</DASUsername>
+        <DASUsername>admin</DASUsername>
         <!-- Administrator password to login to the remote DAS server. -->
-        <DASPassword>${admin.password}</DASPassword>
+        <DASPassword>admin</DASPassword>
 
         <!-- For APIM implemented Statistic client for RDBMS -->
         <StatsProviderImpl>org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl</StatsProviderImpl>
 
         <!-- DAS REST API configuration -->
-        <DASRestApiURL>https://localhost:9444</DASRestApiURL>
-        <DASRestApiUsername>${admin.username}</DASRestApiUsername>
-        <DASRestApiPassword>${admin.password}</DASRestApiPassword>
+        <DASRestApiURL>https://analytics:9444</DASRestApiURL>
+        <DASRestApiUsername>admin</DASRestApiUsername>
+        <DASRestApiPassword>admin</DASRestApiPassword>
 
         <!-- Below property is used to skip trying to connect to event receiver nodes when publishing events even if
             the stats enabled flag is set to true. -->
@@ -216,7 +216,7 @@
              -If you need to start two API Manager instances in the same machine, you need to give different ports to "ThriftServerPort" value in two nodes.
              -ThriftServerHost - Allows to configure a hostname for the thrift server. It uses the carbon hostname by default.
             -The Gateway uses this parameter to connect to the key validation thrift service. -->
-        <KeyValidatorClientType>ThriftClient</KeyValidatorClientType>
+        <KeyValidatorClientType>WSClient</KeyValidatorClientType>
         <ThriftClientConnectionTimeOut>10000</ThriftClientConnectionTimeOut>
         <!--ThriftClientPort>10397</ThriftClientPort-->
 
 No link to traffic-manager
@@ -462,10 +462,10 @@
     <ThrottlingConfigurations>
         <EnableAdvanceThrottling>true</EnableAdvanceThrottling>
         <DataPublisher>
-            <Enabled>true</Enabled>
+            <Enabled>false</Enabled>
             <Type>Binary</Type>
-            <ReceiverUrlGroup>tcp://${carbon.local.ip}:${receiver.url.port}</ReceiverUrlGroup>
-            <AuthUrlGroup>ssl://${carbon.local.ip}:${auth.url.port}</AuthUrlGroup>
+            <ReceiverUrlGroup>tcp://traffic-manager:9611</ReceiverUrlGroup>
+            <AuthUrlGroup>ssl://traffic-manager:9711</AuthUrlGroup>
             <Username>${admin.username}</Username>
             <Password>${admin.password}</Password>
             <DataPublisherPool>
@@ -479,7 +479,7 @@
             </DataPublisherThreadPool>
         </DataPublisher>
         <PolicyDeployer>
-            <ServiceURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServiceURL>
+            <ServiceURL>https://traffic-manager:9443/services/</ServiceURL>
             <Username>${admin.username}</Username>
             <Password>${admin.password}</Password>
         </PolicyDeployer>
@@ -489,8 +489,8 @@
             <Period>3600000</Period-->
         </BlockCondition>
         <JMSConnectionDetails>
-            <Enabled>true</Enabled>
-            <ServiceURL>tcp://${carbon.local.ip}:${jms.port}</ServiceURL>
+            <Enabled>false</Enabled>
+            <ServiceURL>tcp://traffic-manager:5672</ServiceURL>
             <Username>${admin.username}</Username>
             <Password>${admin.password}</Password>
             <Destination>throttleData</Destination>

repository/resources/security/client-truststore.jks, repository/resources/security/wso2cabon.jks
================================================================================================
    Content has been changed but cannot read.


##########################
# wso2am-gateway_manager #
##########################


repository/conf/axis2/axis2.xml
===============================

What does following changes do?
@@ -543,7 +543,7 @@
                 <Password>wso2carbon</Password>
             </TrustStore>
         </parameter>
-        <!--<parameter name="HostnameVerifier">DefaultAndLocalhost</parameter>-->
+        <parameter name="HostnameVerifier">AllowAll</parameter>
             <!--supports Strict|AllowAll|DefaultAndLocalhost or the default if none specified -->
     </transportSender>
 
@@ -628,7 +628,7 @@
      getting this node to join the cluster.
      -->
     <clustering class="org.wso2.carbon.core.clustering.hazelcast.HazelcastClusteringAgent"
-                enable="false">
+                enable="true">
 
         <!--
            This parameter indicates whether the cluster has to be automatically initalized
@@ -650,7 +650,7 @@
                     is deemed to have left the cluster, it will be detected by the Group Membership
                     Service (GMS) using a TCP ping mechanism.
         -->
-        <parameter name="membershipScheme">multicast</parameter>
+        <parameter name="membershipScheme">wka</parameter>
         <!--<parameter name="licenseKey">xxx</parameter>-->
         <!--<parameter name="mgtCenterURL">http://localhost:8081/mancenter/</parameter>-->
 
 repository/conf/api-manager.xml
 ===============================

 @@ -136,20 +136,20 @@
 
              Ex - Multiple Receiver Groups with two receivers each
              {tcp://localhost:7612/,tcp://localhost:7613},{tcp://localhost:7712/,tcp://localhost:7713/} -->
-        <DASServerURL>{tcp://localhost:7612}</DASServerURL>
-        <!--DASAuthServerURL>{ssl://localhost:7712}</DASAuthServerURL-->
+        <DASServerURL>tcp://analytics:7612</DASServerURL>
+        <DASAuthServerURL>ssl://analytics:7712</DASAuthServerURL>
         <!-- Administrator username to login to the remote DAS server. -->
-        <DASUsername>${admin.username}</DASUsername>
+        <DASUsername>admin</DASUsername>
         <!-- Administrator password to login to the remote DAS server. -->
-        <DASPassword>${admin.password}</DASPassword>
+        <DASPassword>admin</DASPassword>
 
         <!-- For APIM implemented Statistic client for RDBMS -->
         <StatsProviderImpl>org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl</StatsProviderImpl>
 
         <!-- DAS REST API configuration -->
-        <DASRestApiURL>https://localhost:9444</DASRestApiURL>
-        <DASRestApiUsername>${admin.username}</DASRestApiUsername>
-        <DASRestApiPassword>${admin.password}</DASRestApiPassword>
+        <DASRestApiURL>https://analytics:9444</DASRestApiURL>
+        <DASRestApiUsername>admin</DASRestApiUsername>
+        <DASRestApiPassword>admin</DASRestApiPassword>
 
         <!-- Below property is used to skip trying to connect to event receiver nodes when publishing events even if
             the stats enabled flag is set to true. -->
@@ -202,7 +202,7 @@
     -->
     <APIKeyValidator>
         <!-- Server URL of the API key manager -->
-        <ServerURL>https://localhost:${mgt.transport.https.port}${carbon.context}services/</ServerURL>
+        <ServerURL>https://keymanager:9443/services/</ServerURL>
 
         <!-- Admin username for API key manager. -->
         <Username>${admin.username}</Username>
@@ -216,7 +216,7 @@
              -If you need to start two API Manager instances in the same machine, you need to give different ports to "ThriftServerPort" value in two nodes.
              -ThriftServerHost - Allows to configure a hostname for the thrift server. It uses the carbon hostname by default.
             -The Gateway uses this parameter to connect to the key validation thrift service. -->
-        <KeyValidatorClientType>ThriftClient</KeyValidatorClientType>
+        <KeyValidatorClientType>WSClient</KeyValidatorClientType>
         <ThriftClientConnectionTimeOut>10000</ThriftClientConnectionTimeOut>
         <!--ThriftClientPort>10397</ThriftClientPort-->
 
@@ -460,7 +460,7 @@
         </WhiteListedURIs>
     </RESTAPI>
     <ThrottlingConfigurations>
-        <EnableAdvanceThrottling>true</EnableAdvanceThrottling>
+        <EnableAdvanceThrottling>false</EnableAdvanceThrottling>
         <DataPublisher>
             <Enabled>true</Enabled>
             <Type>Binary</Type>

repository/deployment/server/synapse-configs/default/api/_AuthorizeAPI_.xml
===========================================================================

@@ -5,7 +5,7 @@
        <property name="uri.var.hostname" expression="get-property('keyManager.hostname')"/>
             <send>
                 <endpoint>
-                    <http uri-template="https://{uri.var.hostname}:{uri.var.portnum}/oauth2/authorize">
+                    <http uri-template="https://keymanager:9443/oauth2/authorize">
                         <timeout>
                             <duration>60000</duration>
                             <responseAction>fault</responseAction>

repository/deployment/server/synapse-configs/default/api/_RevokeAPI_.xml
========================================================================

@@ -5,7 +5,7 @@
        <property name="uri.var.hostname" expression="get-property('keyManager.hostname')"/>
             <send>
                 <endpoint>
-                    <http uri-template="https://{uri.var.hostname}:{uri.var.portnum}/oauth2/revoke">
+                    <http uri-template="https://keymanager:9443/oauth2/revoke">
                         <timeout>
                             <duration>60000</duration>
                             <responseAction>fault</responseAction>

repository/deployment/server/synapse-configs/default/api/_TokenAPI_.xml
=======================================================================

@@ -5,7 +5,7 @@
        <property name="uri.var.hostname" expression="get-property('keyManager.hostname')"/>
             <send>
                 <endpoint>
-                     <http uri-template="https://{uri.var.hostname}:{uri.var.portnum}/oauth2/token">
+                     <http uri-template="https://keymanager:9443/oauth2/token">
                         <timeout>
                             <duration>60000</duration>
                             <responseAction>fault</responseAction>

repository/deployment/server/synapse-configs/default/api/_UserInfoAPI_.xml
==========================================================================

@@ -5,7 +5,7 @@
        <property name="uri.var.hostname" expression="get-property('keyManager.hostname')"/>
             <send>
                 <endpoint>
-                    <http uri-template="https://{uri.var.hostname}:{uri.var.portnum}/oauth2/userinfo">
+                    <http uri-template="https://keymanager:9443/oauth2/userinfo">
                         <timeout>
                             <duration>60000</duration>
                             <responseAction>fault</responseAction>

repository/resources/security/client-truststore.jks, repository/resources/security/wso2cabon.jks
================================================================================================
    Content has been changed but cannot read.


#####################
# wso2am_analytics- #
#####################


repository/conf/tomcat/catalina-server.xml
==========================================

Why below change?
@@ -27,6 +27,7 @@
         -->
         <Connector protocol="org.apache.coyote.http11.Http11NioProtocol"
                    port="9763"
+                   proxyPort="9764"
                    redirectPort="9443"
                    bindOnInit="false"
                    maxHttpHeaderSize="8192"
@@ -52,6 +53,7 @@
         -->
         <Connector protocol="org.apache.coyote.http11.Http11NioProtocol"
                    port="9443"
+                   proxyPort="9444"
                    bindOnInit="false"
                    sslProtocol="TLS"
                    sslEnabledProtocols="TLSv1,TLSv1.1,TLSv1.2"

repository/resources/security/client-truststore.jks, repository/resources/security/wso2cabon.jks
================================================================================================
    Content has been changed but cannot read.